extensions/ql-vscode/src/databases/database-fetcher.ts

```diff
@@ -42,605 +41,566 @@ import { createFilenameFromString } from "../common/filenames";
+   */
+  public async promptImportInternetDatabase(
+    progress: ProgressCallback,
+  ): Promise<DatabaseItem | undefined> {
+    const databaseUrl = await window.showInputBox({
+      prompt: "Enter URL of zipfile of database to download",
+    });
+    if (!databaseUrl) {
+      return;
+    }
+
+    this.validateUrl(databaseUrl);
+
+    const item = await this.databaseArchiveFetcher(
+      databaseUrl,
+      {},
+      undefined,
+      {
+        type: "url",
+        url: databaseUrl,
+      },
+      progress,
     );
-  }
-  return item;
-}
 
-/**
- * Prompts a user to fetch a database from GitHub.
- * User enters a GitHub repository and then the user is asked which language
- * to download (if there is more than one)
- *
- * @param app the App
- * @param databaseManager the DatabaseManager
- * @param storagePath where to store the unzipped database.
- * @param progress the progress callback
- * @param cli the CodeQL CLI server
- * @param language the language to download. If undefined, the user will be prompted to choose a language.
- * @param makeSelected make the new database selected in the databases panel (default: true)
- * @param addSourceArchiveFolder whether to add a workspace folder containing the source archive to the workspace
- */
-export async function promptImportGithubDatabase(
-  app: App,
-  databaseManager: DatabaseManager,
-  storagePath: string,
-  progress: ProgressCallback,
-  cli: CodeQLCliServer,
-  language?: string,
-  makeSelected = true,
-  addSourceArchiveFolder = addDatabaseSourceToWorkspace(),
-): Promise<DatabaseItem | undefined> {
-  const githubRepo = await askForGitHubRepo(progress);
-  if (!githubRepo) {
-    return;
+    if (item) {
+      await this.app.commands.execute("codeQLDatabases.focus");
+      void showAndLogInformationMessage(
+        extLogger,
+        "Database downloaded and imported successfully.",
+      );
+    }
+    return item;
   }
 
-  const databaseItem = await downloadGitHubDatabase(
-    githubRepo,
-    app,
-    databaseManager,
-    storagePath,
-    progress,
-    cli,
-    language,
-    makeSelected,
-    addSourceArchiveFolder,
-  );
-
-  if (databaseItem) {
-    if (makeSelected) {
-      await app.commands.execute("codeQLDatabases.focus");
+  /**
+   * Prompts a user to fetch a database from GitHub.
+   * User enters a GitHub repository and then the user is asked which language
+   * to download (if there is more than one)
+   *
+   * @param progress the progress callback
+   * @param language the language to download. If undefined, the user will be prompted to choose a language.
+   * @param makeSelected make the new database selected in the databases panel (default: true)
+   * @param addSourceArchiveFolder whether to add a workspace folder containing the source archive to the workspace
+   */
+  public async promptImportGithubDatabase(
+    progress: ProgressCallback,
+    language?: string,
+    suggestedRepoNwo?: string,
```

Comment:
The JSDoc above is missing a comment for this parameter, which we should probably add to be consistent.